pkg/webhooks/server.go

```diff
@@ -360,7 +374,61 @@ func (ws *WebhookServer) resourceMutation(request *v1beta1.AdmissionRequest) *v1
 
 func (ws *WebhookServer) resourceValidation(request *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
 	logger := ws.log.WithName("resourceValidation").WithValues("uid", request.UID, "kind", request.Kind.Kind, "namespace", request.Namespace, "name", request.Name, "operation", request.Operation)
+	checked, err := userinfo.IsRoleAuthorize(ws.rbLister, ws.crbLister, ws.rLister, ws.crLister, request)
+	if err != nil {
+		logger.Error(err, "failed to get RBAC infromation for request")
+	}
+
+	if !checked {
+		if request.Operation == v1beta1.Delete || request.Operation == v1beta1.Update {
+			// convert RAW to unstructured
+			var resource *unstructured.Unstructured
+			if request.Operation == v1beta1.Delete {
+				resource, err = enginutils.ConvertToUnstructured(request.OldObject.Raw)
+			} else {
+				resource, err = enginutils.ConvertToUnstructured(request.Object.Raw)
+			}
+			if err != nil {
+				//TODO: skip applying the admission control ?
+				logger.Error(err, "failed to convert RAR resource to unstructured format")
 
+				return &v1beta1.AdmissionResponse{
+					Allowed: false,
+					Result: &metav1.Status{
+						Status:  "Failure",
+						Message: err.Error(),
+					},
+				}
+			}
+
+			oldResource, err := ws.client.GetResource(resource.GetKind(), resource.GetNamespace(), resource.GetName())
+			if err != nil {
+				if !apierrors.IsNotFound(err) {
+					logger.Error(err, "failed to get resource")
+					return &v1beta1.AdmissionResponse{
+						Allowed: false,
+						Result: &metav1.Status{
+							Status:  "Failure",
+							Message: err.Error(),
+						},
+					}
+				}
+			}
+			labels := oldResource.GetLabels()
+			if labels != nil {
+				if labels["app.kubernetes.io/managed-by"] == "kyverno" && labels["app.kubernetes.io/synchronize"] == "enable" {
+					return &v1beta1.AdmissionResponse{
+						Allowed: false,
+						Result: &metav1.Status{
+							Status:  "Failure",
+							Message: "You don't have permission to update resourses that is generated by kyverno",
```

Comment:
Reword the message to "Resource is managed by Kyverno, can't be changed manually. You can edit generate policy to update this resource.'

Reply:
ok